development-practices/unlocked-packages.md

```diff
@@ -34,13 +34,19 @@ The following sections deal with items that are particular to DX@Scale or more e
   * A group of related code and customization
   * Independent from other components and can be called from other packages
   * Standalone and released independently
-* Don’t package the metadata merely because it is supported by Unlocked Packaging. There are many scenarios where if  particular metadata is cross-cutting across different packages \(say layouts\) and packaging them might result in too many dependencies
-* **Custom Labels:** Group and manage custom labels for each package separately to ensure they don't cause deployment errors. sfpowerkit provides some tooling around to work with maintaining custom labels. Check the command [here](https://github.com/Accenture/sfpowerkit#sfpowerkitsourcecustomlabelcreate) and [here](https://github.com/Accenture/sfpowerkit#sfpowerkitsourcecustomlabelreconcile). Ensure you use[ sfpowerkit label create ](https://github.com/Accenture/sfpowerkit#sfpowerkitsourcecustomlabelcreate)command to create these labels.  **Please ensure you use custom label's for its intended purpose as in for support texts in a multilingual app, do not use it to store constants etc.**
-* Care must be taken when dealing with the following metadata. Most often, they must be placed in a source package in the respective domain, or if its a cross-cutting concern, move it to global packages such as src-access-managements/src-ui  
-  * Profiles
-  * Permission Sets
-  * Layouts    
-* **Workflows and ProcessBuilders** need to be placed in the same unlocked package that contains the parent object definition. This is a restriction of this particular metadata component, In these scenarios rather than building  automation using workflows and process builders, it is better to use apex or flow
+* Don’t package the metadata merely because it is supported by Unlocked Packaging. Some components (e.g. profiles) will need to be managed outside of unlocked packages and it may be preferable to deploy other related metadata using the same mechanism. In other cases, metadata like non-critical reports, email templates etc, may be better to manage outside of source control entirely to allow these to be changed dynamically by end users.
+* **Objects** should be "owned", where possible, by one domain specific package which includes the majority of fields, validation rules and other metadata related to this object. Other metadata acting on the object should be scoped where possible to operate within the confines of the package. Examples include:
+  * **Permission Sets**, which should be granular and specific to an object or package
```

Comment:
We run the risk of having very SObject related permission sets, agreed that the permission sets must be granular at the function level, for example: **ability to create customers** which leads to the creation of a permission set that enables the creation/modification of account and contact records or **ability to administrate customers** which would lead to another permission set that would enable delete/viewAll permissions on the same SObject.
Probably also worth mentioning that the permission sets should be composable.

Reply:
Yes good point. I've recommended object specific in orgs I've worked on partly because it's quite a clear way of ensuring granularity but you're right the main point is the granularity and the object-specific aspect is pretty arbitrary. I've updated this point with your examples and a few more details. And 100% on the composable aspect, this is really crucial, I've added a point around this too